Source/ASCollectionView.mm

```diff
@@ -1831,6 +1831,17 @@ - (void)rangeController:(ASRangeController *)rangeController didUpdateWithChange
       
       __block NSUInteger numberOfUpdates = 0;
       [self _superPerformBatchUpdates:^{
+        
+        // Use native moveItem instead of Delete-Insert
+        NSMutableArray *fromIndexPaths = [NSMutableArray array];
+        NSMutableArray *toIndexPaths = [NSMutableArray array];
+        for (_ASHierarchyMoveItemChange *change in [changeSet itemChangesOfType:_ASHierarchyChangeTypeMove]) {
+          [super moveItemAtIndexPath:change.fromIndexPath toIndexPath:change.toIndexPath];
+          [fromIndexPaths addObject:change.fromIndexPath];
+          [toIndexPaths addObject:change.toIndexPath];
+          numberOfUpdates++;
+        }
+        
```

Comment:
To make sure the element maps of `ASDataController` are correctly synced, the steps described by @Adlai-Holler [here](https://github.com/facebook/AsyncDisplayKit/pull/2868#discussion_r94865245) should be implemented in `-ASDataController:_updateElementsInMap:changeSet:`. 

Then in `ASCollectionView` and `ASTableView`, we can simply forward each move update to `super`.